models/bonus.py

```diff
@@ -0,0 +1,212 @@
+    so_line = fields.Many2one(related='timesheet_id.so_line')
+    employee_id = fields.Many2one(related='timesheet_id.employee_id', store=True, required=1, copy=True)
+    order_id = fields.Many2one(related='so_line.order_id', store=True, required=1, copy=True)
+    company_id = fields.Many2one(related='order_id.company_id')
+    currency_id = fields.Many2one(related='company_id.currency_id')
+    amount = fields.Monetary(string='Amount', required=1)
+
+    vendor_bill_move_ids = fields.Many2many(
+        'account.move', compute='_compute_vendor_bill_move_ids',
+        help="Vendor bill but also Vendor bill credit note")
+    vendor_bill_move_count = fields.Integer(string='# Invoices', compute='_compute_vendor_bill_move_count', groups="account.group_account_manager")
+    vendor_bill_move_line_ids = fields.Many2many('account.move.line', ondelete='restrict')
+
+    @api.depends('vendor_bill_move_ids')
+    def _compute_vendor_bill_move_count(self):
+        for bonus in self:
+            bonus.vendor_bill_move_count = len(bonus.vendor_bill_move_ids)
+
+    @api.depends('vendor_bill_move_line_ids')
+    def _compute_vendor_bill_move_ids(self):
+        for bonus in self:
+            bonus.vendor_bill_move_ids = bonus.vendor_bill_move_line_ids.move_id
+
+    def unlink(self):
+        move_lines = self.vendor_bill_move_line_ids
+        res = super().unlink()
+        for line in move_lines:
+            vendor_bill = line.move_id
+            line.unlink()
+            # Unlink the vendor bill if nothing in it anymore
+            if not vendor_bill.invoice_line_ids:
+                vendor_bill.write({'state': 'draft'})
+                vendor_bill.unlink()
+        return res
+
+    def action_view_invoices(self):
+        action = self.env['ir.actions.act_window']._for_xml_id('gse_bonuses.action_view_invoices')
+        # action['display_name'] = self.name
+        action['domain'] = [('id', 'in', self.vendor_bill_move_ids.ids)]
+        context = action['context'].replace('active_id', str(self.id))
+        action['context'] = ast.literal_eval(context)
+        return action
+
+    def generate_bonuses(self, order):
+        if not order:
+            return
+
+        if not (order.date_order > fields.Datetime.from_string('2023-05-31 23:59:59')):
+            # This should not but a `raise UserError()` because you still want
+            # to validate SO after that date but just not generating bonuses.
+            logger.info("Impossible de générer un bonus pour une SO validée avant le 1er juin 2023 (SO %s %s).", order.id, order.date_order)
+            return
+
+        order.ensure_one()
+
+        # Check if bonuses can be created: the SO must:
+        # - be fully paid
+        # - be fully delivered
+        # - have its related labor service tasks done (related SOL will be
+        #   marked as delivered, so it should be covered by previous point)
```

Comment:
Can we add 2 fields and 2 conditions that have just emerged:
1. On an employee's contract, add a boolean: Allow Technician Transport Expenses.
2. On the task, add a boolean: Exclude from Technician Transport Expenses.
If the first is false, do not compute the bonus
If the second is true, do not compute the bonus.

Reply:
![Screenshot 2023-12-21 at 14 01 51](https://github.com/sbuhl/gse_bonuses/assets/505562/8d3e6fe6-d3d2-4921-ab02-9852b5a79aec)
![Screenshot 2023-12-21 at 14 04 01](https://github.com/sbuhl/gse_bonuses/assets/505562/8095a8e7-7891-4c29-98a6-eaff9937518e)

Reply:
Done